org.uqbar.project.wollok.typeSystem/src/org/uqbar/project/wollok/typesystem/constraints/variables/SimpleTypeInfo.xtend

```diff
@@ -57,7 +58,7 @@ class SimpleTypeInfo extends TypeInfo {
 
 	override setMaximalConcreteTypes(MaximalConcreteTypes maxTypes, TypeVariable origin) {
 		minTypesDo(origin) [
-			if (!maxTypes.contains(type))
+			if (!maxTypes.empty && !maxTypes.contains(type))
```

Comment:
I am not sure about max types being empty => it means an error, there is no possible type for this var. In this case, the var should have an error and therefore should be not further analyzed, isn't it?

Reply:
Yeah, fixed! (It was sealing a tvar to Any and that set an empty MaxType).